components/brave_sync/brave_sync_service_impl.cc

```diff
@@ -1069,199 +916,200 @@ void BraveSyncServiceImpl::SendDeviceSyncRecord(const int &action,
+    record->objectId = tools::GenerateObjectId();
+    record->action = jslib::SyncRecord::Action::CREATE;
+
+    DCHECK(bookmark->order.empty());
+    int index = node->parent()->GetIndexOf(node);
+    if (node->is_folder()) {
+      bookmark->order =
+          sync_prefs_->GetBookmarksBaseOrder() +
+          "." +
+          std::to_string(index);
+    } else {
+      std::string order;
+      node->parent()->GetMetaInfo("order", &order);
+      DCHECK(!order.empty());
+      bookmark->order = order + "." + std::to_string(index);
+    }
+  } else {
+    record->action = jslib::SyncRecord::Action::UPDATE;
+    DCHECK(!bookmark->order.empty());
+    DCHECK(!record->objectId.empty());
   }
-}
 
-void BraveSyncServiceImpl::CreateUpdateDeleteBookmarks(
-  const int &action,
-  const std::vector<InitialBookmarkNodeInfo> &list,
-  const std::map<const bookmarks::BookmarkNode*, std::string> &order_map,
-  const bool &addIdsToNotSynced) {
-  LOG(ERROR) << "TAGAB brave_sync::BraveSyncServiceImpl::CreateUpdateDeleteBookmarks";
-  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+  return record;
+}
 
-  LOG(ERROR) << "TAGAB brave_sync::BraveSyncServiceImpl::CreateUpdateDeleteBookmarks list.empty()=" << list.empty();
-  LOG(ERROR) << "TAGAB brave_sync::BraveSyncServiceImpl::CreateUpdateDeleteBookmarks sync_initialized_=" << sync_initialized_;
-  LOG(ERROR) << "TAGAB brave_sync::BraveSyncServiceImpl::CreateUpdateDeleteBookmarks sync_prefs_->GetSyncBookmarksEnabled()=" << sync_prefs_->GetSyncBookmarksEnabled();
 
-  if (list.empty() || !sync_initialized_ || !sync_prefs_->GetSyncBookmarksEnabled() ) {
-    return;
+void BraveSyncServiceImpl::GetExistingBookmarks(
+    const std::vector<std::unique_ptr<jslib::SyncRecord>>& records,
+    SyncRecordAndExistingList* records_and_existing_objects) {
+  for (const auto& record : records){
+    auto resolved_record = std::make_unique<SyncRecordAndExisting>();
+    resolved_record->first = jslib::SyncRecord::Clone(*record);
+    auto* node = FindByObjectId(bookmark_model_, record->objectId);
+    if (node)
+      resolved_record->second = BookmarkNodeToSyncBookmark(node);
+    records_and_existing_objects->push_back(std::move(resolved_record));
   }
-
-  task_runner_->PostTask(
-    FROM_HERE,
-    base::Bind(&BraveSyncServiceImpl::CreateUpdateDeleteBookmarksFileWork,
-               weak_ptr_factory_.GetWeakPtr(),
-               action,
-               list,
-               order_map,
-               addIdsToNotSynced));
 }
 
-void BraveSyncServiceImpl::CreateUpdateDeleteBookmarksFileWork(
-  const int &action,
-  const std::vector<InitialBookmarkNodeInfo> &list,
-  const std::map<const bookmarks::BookmarkNode*, std::string> &order_map,
-  const bool &addIdsToNotSynced) {
-  LOG(ERROR) << "TAGAB brave_sync::BraveSyncServiceImpl::CreateUpdateDeleteBookmarksFileWork";
-  LOG(ERROR) << "TAGAB addIdsToNotSynced=" << addIdsToNotSynced;
+void BraveSyncServiceImpl::SendUnsyncedBookmarks() {
```

Comment:
but aren't they exactly the same set of bookmarks?

Reply:
I guess the problem is that I'm setting the object_id on send, so later it will looked synced even if it isn't. I'll take a look at some changes for that. Thanks!

Reply:
I guess we should check the timestamp it was synced which we can also store in metainfo. wdyt?

Reply:
@bridiver 
>but aren't they exactly the same set of bookmarks?
I think no. For example 
we had bookmarks A.com, B.com, C.com
we enabled sync
all of A/B/C has been sent to sync, as they do not have `objectId` in metainfo.

Another case, B.com was modified to BB.com, we done send-sync-records, but somehow record had not arrived to AWS. BB.com will still have the objectId.

>I guess we should check the timestamp it was synced which we can also store in metainfo. 
in details:
when we send the record, we save metainfos 'last_send_time'+'last_action'
if we see record in `get-existing-objects` we clear both 'last_send_time'+'last_action' .
when we want to resend unsynced records, we go through all the nodes and search for 'last_send_time'.

I think this will work well, except deleted records.